src/CMakeLists.txt

```diff
@@ -295,6 +295,9 @@ elseif(ALLOCATOR STREQUAL "jemalloc")
   set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
 elseif(ALLOCATOR STREQUAL "libc")
   set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
+else()
+  set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
+  message(WARNING "Possibly incorrect allocator selected:" ${ALLOCATOR})
```

Comment:
Can we raise an error here?

Reply:
yes, we should error out here with an error message.

Reply:
Please don't.
FreeBSD has jemalloc as default allocator. 
Defining it however, will show that the additional tools are not available in ways that are expected.
So for example rocksdb starts doing wrong things.
If you want to error out, then please make an exemption for FreeBSD.

Reply:
it's always set in `$top_srcdir/CMakeLists.txt`.

Reply:
You are right. So FreeBSD ends in libc.
Why not error out earlier, like in `$top_srcdir/CMakeLists.txt`?